packages/expect/src/asymmetric_matchers.js

```diff
@@ -218,6 +222,10 @@ class StringMatching extends AsymmetricMatcher {
   }
 
   asymmetricMatch(other: string) {
+    if (!isA('String', other) && !isA('RegExp', other)) {
```

Comment:
Although `sample` can be either string or regexp, it seems like `other` can only be a string.